GitUpKit/Interface/GIGraphView.m

```diff
@@ -860,112 +860,164 @@ static void _DrawLine(GILine* line, CGContextRef context, CGFloat offset, CGFloa
+      _AppendAttributedString((CFMutableAttributedStringRef)multilineTitle, [NSString stringWithFormat:@"%@\n", localBranchName], multilineTitleAttributes);
+      [multilineTitle addAttribute:NSFontAttributeName value:(id)boldFont range:localBranchNameRange];
+      [multilineTitle addAttribute:NSForegroundColorAttributeName value:darkColor range:localBranchNameRange];
+      
+      // Upstream may be local or remote, so we look for slash to make the branch name more emphasized
+      NSRange upstreamBranchNameRange = NSMakeRange(multilineTitle.length + 2, upstreamName.length); // +2 to exclude "⬅ "
+      NSRange slashRange = [upstreamName rangeOfString:@"/"];
+      if (slashRange.location != NSNotFound) {
+        upstreamBranchNameRange.location += NSMaxRange(slashRange);
+        upstreamBranchNameRange.length -= NSMaxRange(slashRange);
+      }
+      _AppendAttributedString((CFMutableAttributedStringRef)multilineTitle, [NSString stringWithFormat:@"⬅ %@\n", upstreamName], multilineTitleAttributes);
+      [multilineTitle addAttribute:NSForegroundColorAttributeName value:darkColor range:upstreamBranchNameRange];
+      
     } else {
-      [label appendString:localBranch.name];
+      NSString* branchName = localBranch.name;
+      NSRange branchNameRange = NSMakeRange(multilineTitle.length, branchName.length);
+      _AppendAttributedString((CFMutableAttributedStringRef)multilineTitle, [NSString stringWithFormat:@"%@\n", branchName], multilineTitleAttributes);
+      [multilineTitle addAttribute:NSFontAttributeName value:(id)boldFont range:branchNameRange];
+      [multilineTitle addAttribute:NSForegroundColorAttributeName value:darkColor range:branchNameRange];
     }
-    ++index;
+    
+    _AppendAttributedString((CFMutableAttributedStringRef)multilineTitle, @"\n", nil);
   }
-  index = 0;
+  
   for (GCHistoryRemoteBranch* remoteBranch in branch.remoteBranches) {
-    if (index) {
-      [label appendString:@", "];
-    } else if (label.length) {
-      [label appendString:@"\n"];
-    }
-    [label appendString:remoteBranch.name];
-    ++index;
+    NSString* branchName = remoteBranch.branchName;
+    _AppendAttributedString((CFMutableAttributedStringRef)multilineTitle, [NSString stringWithFormat:@"%@\n", remoteBranch.name], multilineTitleAttributes);
+    NSRange branchNameRange = NSMakeRange(multilineTitle.length - branchName.length - 1, branchName.length); // -1 to exclude '\n'
+    [multilineTitle addAttribute:NSFontAttributeName value:(id)boldFont range:branchNameRange];
+    [multilineTitle addAttribute:NSForegroundColorAttributeName value:darkColor range:branchNameRange];
   }
-  index = 0;
-  for (GCHistoryTag* tag in branch.tags) {
-    if (index) {
-      [label appendString:@", "];
-    } else {
-      if (label.length) {
-        [label appendString:@"\n"];
-      }
-      [label appendString:@"["];
-    }
-    [label appendString:tag.name];
-    ++index;
+  
+  if (branch.remoteBranches.count > 0) {
```

Comment:
This should be `(branch.remoteBranches.count && branch.tags.count)` so that you only insert a newline if there are branches and tags. Then get rid of `_AppendAttributedString((CFMutableAttributedStringRef)multilineTitle, @"\n", nil);` further down.